src/pip/_internal/req/constructors.py

```diff
@@ -236,26 +289,9 @@ def install_req_from_line(
         link = Link(name)
     else:
         p, extras_as_string = _strip_extras(path)
-        looks_like_dir = os.path.isdir(p) and (
-            os.path.sep in name or
-            (os.path.altsep is not None and os.path.altsep in name) or
-            name.startswith('.')
-        )
-        if looks_like_dir:
-            if not is_installable_dir(p):
-                raise InstallationError(
-                    "Directory %r is not installable. Neither 'setup.py' "
-                    "nor 'pyproject.toml' found." % name
-                )
-            link = Link(path_to_url(p))
-        elif is_archive_file(p):
-            if not os.path.isfile(p):
-                logger.warning(
-                    'Requirement %r looks like a filename, but the '
-                    'file does not exist',
-                    name
-                )
-            link = Link(path_to_url(p))
+        path_or_url = _get_path_to_url(p, name)
```

Comment:
Sorry if I didn't say this before, but I think what you want here is `url = _get_url_from_path(p, name)` since you know the return value will be a url (if non-None). And then below you want `if url is not None:` since you are using `None` as your non-url return value.

Reply:
Got it. About the `is not None`, it feels kind of confusing to me.`if url: ` would have the same effect as `if url is not None:`, right?

Reply:
More than likely, yes. 😄 But it's best to be clear / exact about how your code behaves if you know it's written a certain way, and `None` is exactly what you want.